### PR TITLE
[ZEPPELIN-1313] NullPointerException when using Clone notebook REST API

### PR DESCRIPTION
**Pull request by kavinkumarks**
### What is this PR for?
This fixes when the input json is empty for the clone notebook REST API and for this case the default name will be populated.

### What type of PR is it?
Bug Fix

### Todos
NA

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1313

### How should this be tested?
When the input json is empty for the clone notebook REST API, the response should be 201 with the default name populated.

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by kavinkumarks** [comment]
Reopening to trigger the CI build.

**Comment by kavinkumarks** [comment]
Reopening to trigger the CI build.

**Comment by kavinkumarks** [comment]
The CI build is green refer to https://travis-ci.org/apache/zeppelin/builds/154093445

Could someone please review this?

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java [hunk 1]
```diff
@@ -239,6 +239,8 @@ public Note cloneNote(String sourceNoteID, String newNoteName, AuthenticationInf
     Note newNote = createNote(subject);
     if (newNoteName != null) {
       newNote.setName(newNoteName);
+    } else {
+      newNote.setName("Note " + newNote.getId());
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: just `newNote.id()`?

**Comment by felixcheung** [comment]
LGTM

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java (hunk 1 above)
Comment (jongyoul) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: Those methods returns the same. I think we need to remove `note.id()`.

**Comment by jongyoul** [comment]
LGTM

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java (hunk 1 above)
Comment (kavinkumarks) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: @felixcheung yes we use only note.getId() even in other place like create notebook.
@jongyoul I will remove note.id() with this.

Comment (jongyoul) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: @kavinkumarks `note.id()` is used widely and it's better to handle with another PR.

Comment (kavinkumarks) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java]: @jongyoul I did the changes already with this ticket and running CI build now.Please let me know if I need to revert them.

**Comment by jongyoul** [comment]
@kavinkumarks Thanks for the removing the method, but personally, I think we should handle it with another PR. @felixcheung What do you think of it? It looks irrelevant.

**Comment by felixcheung** [comment]
I agree, that should be a separate PR/JIRA.

I'm ok with including it now since changes are already made? Is CI passing?

**Comment by jongyoul** [comment]
@felixcheung Yeah, @kavinkumarks already did it, thus I also am OK to include it. CI failure is not related to this PR. LGTM

**Comment by kavinkumarks** [comment]
@felixcheung / @jongyoul could you please take care of merging this?

Thanks,
Kavin

**Comment by kavinkumarks** [comment]
@felixcheung / @jongyoul any updates on this?

Thanks,
Kavin

**Comment by jongyoul** [comment]
@AhyoungRyu Could you please merge this PR? [This](https://cwiki.apache.org/confluence/display/ZEPPELIN/Committer+Guide) will help you

**Comment by felixcheung** [comment]
I could also merge this tomorrow if you are having problems.

Sorry about the delay.

**Comment by jongyoul** [comment]
@felixcheung I don't have any problem. I'd like to give @AhyoungRyu a chance to merge PR for the first time. :-)

**Comment by felixcheung** [comment]
Right, when I say "you" I mean Ahyoung :)

**Comment by AhyoungRyu** [comment]
@jongyoul @felixcheung Sure. Let me try :D

**Comment by kavinkumarks** [comment]
Thanks everyone!
- Kavin

**Pull request closed without merging**